models/vrp.rb

```diff
@@ -123,6 +123,7 @@ def self.create(hash, delete = true)
       vrp = super({})
       self.check_consistency(hash)
       self.ensure_retrocompatibility(hash)
+      self.deduce_data_from_hash(hash)
```

Comment:
Is it possible to move this logic inside `expand_data`?

Reply:
The idea was not to have _direct_ field within shipments model. In expand_data we would not have this information (_direct_) anymore so it is not possible to move this logic if we keep it like this.

I am not sure it is worth to have _direct_ field in _shipments_ model because we can just generate corresponding relation if we need it. But, as you said it, in heuristics it could be useful to have a "shortcut" to know wether it is a direct shipment or not. 

What do you think ?